File: zmaticoo-ios-demo/SettingViewController.m
//
//  SettingViewController.m
//  zmaticoo-ios-demo
//

#import "SettingViewController.h"
#import <MaticooSDK/MaticooSDK.h>

@interface SettingViewController ()
@property (nonatomic, strong) UISwitch *switchGdpr;
@property (nonatomic, strong) UISwitch *switchDoNotStatus;
@property (nonatomic, strong) UISwitch *switchCoppa;
@end

@implementation SettingViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Setting";
    self.view.backgroundColor = [self groupedBackgroundColor];
    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone
                                                                                           target:self
                                                                                           action:@selector(doneTapped)];

    UIScrollView *scroll = [[UIScrollView alloc] initWithFrame:CGRectZero];
    scroll.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:scroll];

    UIStackView *stack = [[UIStackView alloc] init];
    stack.axis = UILayoutConstraintAxisVertical;
    stack.spacing = 16;
    stack.translatesAutoresizingMaskIntoConstraints = NO;
    [scroll addSubview:stack];

    UILayoutGuide *safe = self.view.safeAreaLayoutGuide;
    [NSLayoutConstraint activateConstraints:@[
        [scroll.topAnchor constraintEqualToAnchor:safe.topAnchor],
        [scroll.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [scroll.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [scroll.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor],
        [stack.topAnchor constraintEqualToAnchor:scroll.contentLayoutGuide.topAnchor constant:16],
        [stack.leadingAnchor constraintEqualToAnchor:scroll.contentLayoutGuide.leadingAnchor constant:16],
        [stack.trailingAnchor constraintEqualToAnchor:scroll.contentLayoutGuide.trailingAnchor constant:-16],
        [stack.bottomAnchor constraintEqualToAnchor:scroll.contentLayoutGuide.bottomAnchor constant:-16],
        [stack.widthAnchor constraintEqualToAnchor:scroll.frameLayoutGuide.widthAnchor constant:-32],
    ]];

    self.switchGdpr = [[UISwitch alloc] init];
    self.switchDoNotStatus = [[UISwitch alloc] init];
    self.switchCoppa = [[UISwitch alloc] init];

    MaticooAds *sdk = [MaticooAds shareSDK];
    self.switchGdpr.on = [sdk getConsentStatus];
    self.switchDoNotStatus.on = [sdk getDoNotSell];
    self.switchCoppa.on = [sdk getIsAgeRestrictedUser];

    [self.switchGdpr addTarget:self action:@selector(gdprChanged:) forControlEvents:UIControlEventValueChanged];
    [self.switchDoNotStatus addTarget:self action:@selector(doNotStatusChanged:) forControlEvents:UIControlEventValueChanged];
    [self.switchCoppa addTarget:self action:@selector(coppaChanged:) forControlEvents:UIControlEventValueChanged];

    [stack addArrangedSubview:[self rowWithTitle:@"GDPR" switchView:self.switchGdpr]];
    [stack addArrangedSubview:[self rowWithTitle:@"DoNotStatus" switchView:self.switchDoNotStatus]];
    [stack addArrangedSubview:[self rowWithTitle:@"CoppaStatus" switchView:self.switchCoppa]];
}

- (UIColor *)groupedBackgroundColor {
    if (@available(iOS 13.0, *)) {
        return [UIColor systemGroupedBackgroundColor];
    }
    return [UIColor groupTableViewBackgroundColor];
}

- (UIView *)rowWithTitle:(NSString *)title switchView:(UISwitch *)sw {
    UIView *row = [[UIView alloc] init];
    row.backgroundColor = [self cardColor];
    row.layer.cornerRadius = 8;
    row.layer.masksToBounds = YES;

    UILabel *label = [[UILabel alloc] init];
    label.text = title;
    label.font = [UIFont systemFontOfSize:16];
    label.translatesAutoresizingMaskIntoConstraints = NO;

    sw.translatesAutoresizingMaskIntoConstraints = NO;

    [row addSubview:label];
    [row addSubview:sw];

    [NSLayoutConstraint activateConstraints:@[
        [label.leadingAnchor constraintEqualToAnchor:row.leadingAnchor constant:12],
        [label.centerYAnchor constraintEqualToAnchor:row.centerYAnchor],
        [label.trailingAnchor constraintLessThanOrEqualToAnchor:sw.leadingAnchor constant:-8],
        [sw.trailingAnchor constraintEqualToAnchor:row.trailingAnchor constant:-12],
        [sw.centerYAnchor constraintEqualToAnchor:row.centerYAnchor],
        [row.heightAnchor constraintGreaterThanOrEqualToConstant:44],
    ]];

    return row;
}

- (UIColor *)cardColor {
    if (@available(iOS 13.0, *)) {
        return [UIColor secondarySystemGroupedBackgroundColor];
    }
    return [UIColor whiteColor];
}

- (void)doneTapped {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)gdprChanged:(UISwitch *)sender {
    [[MaticooAds shareSDK] setConsentStatus:sender.isOn];
}

- (void)doNotStatusChanged:(UISwitch *)sender {
    [[MaticooAds shareSDK] setDoNotSell:sender.isOn];
}

- (void)coppaChanged:(UISwitch *)sender {
    [[MaticooAds shareSDK] setIsAgeRestrictedUser:sender.isOn];
}

@end

File: zmaticoo-ios-demo-swift/SettingViewController.swift
//
//  SettingViewController.swift
//  zmaticoo-ios-demo-swift
//
//  Swift 版隐私设置页：对齐 ObjC 版 SettingViewController
//  （GDPR / DoNotSell / COPPA 三个开关，实时写入 SDK）。
//

import UIKit
import MaticooSDK

final class SettingViewController: UIViewController {

    private let switchGdpr = UISwitch()
    private let switchDoNotSell = UISwitch()
    private let switchCoppa = UISwitch()

    override func viewDidLoad() {
        super.viewDidLoad()
        title = "Setting"
        view.backgroundColor = DemoTheme.groupedBackgroundColor
        navigationItem.rightBarButtonItem = UIBarButtonItem(
            barButtonSystemItem: .done, target: self, action: #selector(doneTapped)
        )

        let scroll = UIScrollView()
        scroll.translatesAutoresizingMaskIntoConstraints = false
        view.addSubview(scroll)

        let stack = UIStackView()
        stack.axis = .vertical
        stack.spacing = 16
        stack.translatesAutoresizingMaskIntoConstraints = false
        scroll.addSubview(stack)

        NSLayoutConstraint.activate([
            scroll.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            scroll.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            scroll.trailingAnchor.constraint(equalTo: view.trailingAnchor),
            scroll.bottomAnchor.constraint(equalTo: view.bottomAnchor),
            stack.topAnchor.constraint(equalTo: scroll.contentLayoutGuide.topAnchor, constant: 16),
            stack.leadingAnchor.constraint(equalTo: scroll.contentLayoutGuide.leadingAnchor, constant: 16),
            stack.trailingAnchor.constraint(equalTo: scroll.contentLayoutGuide.trailingAnchor, constant: -16),
            stack.bottomAnchor.constraint(equalTo: scroll.contentLayoutGuide.bottomAnchor, constant: -16),
            stack.widthAnchor.constraint(equalTo: scroll.frameLayoutGuide.widthAnchor, constant: -32),
        ])

        let sdk = MaticooAds.shareSDK()
        switchGdpr.isOn = sdk.getConsentStatus()
        switchDoNotSell.isOn = sdk.getDoNotSell()
        switchCoppa.isOn = sdk.getIsAgeRestrictedUser()

        switchGdpr.addTarget(self, action: #selector(gdprChanged(_:)), for: .valueChanged)
        switchDoNotSell.addTarget(self, action: #selector(doNotSellChanged(_:)), for: .valueChanged)
        switchCoppa.addTarget(self, action: #selector(coppaChanged(_:)), for: .valueChanged)

        stack.addArrangedSubview(row(title: "GDPR", switchView: switchGdpr))
        stack.addArrangedSubview(row(title: "DoNotSell", switchView: switchDoNotSell))
        stack.addArrangedSubview(row(title: "CoppaStatus", switchView: switchCoppa))
    }

    @objc private func doneTapped() {
        dismiss(animated: true, completion: nil)
    }

    @objc private func gdprChanged(_ sender: UISwitch) {
        MaticooAds.shareSDK().setConsentStatus(sender.isOn)
    }

    @objc private func doNotSellChanged(_ sender: UISwitch) {
        MaticooAds.shareSDK().setDoNotSell(sender.isOn)
    }

    @objc private func coppaChanged(_ sender: UISwitch) {
        MaticooAds.shareSDK().setIsAgeRestrictedUser(sender.isOn)
    }

    // MARK: - UI Helpers

    private func row(title: String, switchView: UISwitch) -> UIView {
        let row = UIView()
        row.backgroundColor = DemoTheme.cardBackgroundColor
        row.layer.cornerRadius = 8
        row.layer.masksToBounds = true

        let label = UILabel()
        label.text = title
        label.font = UIFont.systemFont(ofSize: 16)
        label.translatesAutoresizingMaskIntoConstraints = false
        switchView.translatesAutoresizingMaskIntoConstraints = false

        row.addSubview(label)
        row.addSubview(switchView)

        NSLayoutConstraint.activate([
            label.leadingAnchor.constraint(equalTo: row.leadingAnchor, constant: 12),
            label.centerYAnchor.constraint(equalTo: row.centerYAnchor),
            label.trailingAnchor.constraint(lessThanOrEqualTo: switchView.leadingAnchor, constant: -8),
            switchView.trailingAnchor.constraint(equalTo: row.trailingAnchor, constant: -12),
            switchView.centerYAnchor.constraint(equalTo: row.centerYAnchor),
            row.heightAnchor.constraint(greaterThanOrEqualToConstant: 44),
        ])
        return row
    }
}
